text/0000-route-model-name-property.md

```diff
@@ -0,0 +1,149 @@
+
+```
+app/routes/posts.js
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  model() {
+    return this.store.findAll('post');
+  },
+  setupController(model, controller){
+    controller.set('articles', model);
+  }
+});
+```
+
+This RFC proposes a  solution like this:
+
+```
+app/routes/posts.js
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  model() {
+    return this.store.findAll('post');
+  },
+  modelName: articles
+});
+```
+
+- Add a property called `modelName` to the Route API. When the controller is initialised, the return value from the routes `model()` hook is assigned to the controller with value of the `modelName` property as the key.
+- This change would not require us to stop assigning the return value to `model` key on the controller as well.
+- Generator could be updated so that `ember generate route foo` would add `modelName: foo` to the generated route file.
+
+## How we teach this
+
+> What names and terminology work best for these concepts and why? How is this
+idea best presented? As a continuation of existing Ember patterns, or as a
+wholly new one?
+
+This doesn't introduce any new concepts or patterns to Ember, however this API is more explicit and user friendly than the current behaviour. At the moment the fact that the return from the `model()` hook on the route will be assigned to the controllers `model` key is _ember magic™️_, which must be taught.
+
+> Would the acceptance of this proposal mean the Ember guides must be
+re-organized or altered? Does it change how Ember is taught to new users
+at any level?
+
+We would continue to mention the application of the model hook in the guides as we currently do:
+
+> [_"The route will then set the return value from the model hook as the model property of the controller."_](https://guides.emberjs.com/release/routing/specifying-a-routes-model/)
+
+We could add an extra sentence covering this feature:
+
+You can also set the return value from the model hook to be any custom property name on the controller by using `modelName` property in the route.
+```
+app/routes/posts.js
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  model() {
+    return this.store.findAll('post');
+  },
+  modelName: articles
+});
+```
+```
+app/templates/posts.hbs
+<h1>Articles</h1>
+{{#each this.articles as |article|}}
+  <p>{{articles.body}}</p>
+{{/each}}
+```
+> How should this feature be introduced and taught to existing Ember
+users?
+
+We can let developers know that instead of doing:
+```
+app/routes/posts.js
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  model() {
+    return this.store.findAll('post');
+  },
+  setupController(model, controller){
+    controller.set('customModelName', model);
+  }
+});
+```
+They can simply use  the `modelName` property on the route, if they want to 😀.
```

Comment:
```suggestion
They can use the `modelName` property on the route, if they want to.
```

Reply:
My space bar  is broken 😂

Reply:
Also I am going to start campaigning for more Emojis in RFCs.